src/bin/test.rs

```diff
@@ -98,10 +101,7 @@ pub fn execute(options: Options, config: &Config) -> CliResult<Option<()>> {
     match err {
         None => Ok(None),
         Some(err) => {
-            Err(match err.exit.as_ref().and_then(|e| e.code()) {
-                Some(i) => CliError::new("", i),
```

Comment:
This is actually something that Cargo needs to keep doing (propagating the same exit code as a test) in the case that we bailed out after the first test, would it be possible to ferry along that error code up here?